flexget/plugins/metainfo/nfo_lookup.py

```diff
@@ -0,0 +1,226 @@
+    described in the link below
+    https://docs.python.org/3/library/xml.html#xml-vulnerabilities
+
+    Use this only with nfo files you have created yourself.
+    """
+    schema = {'type': 'boolean'}
+    nfo_file_extension = '.nfo'
+
+    # This priority makes sure this plugin runs before the imdb_lookup plugin, if it is also used. That way setting
+    # imdb_id here will help imdb_lookup find the correct movie.
+    @plugin.priority(150)
+    def on_task_metainfo(self, task, config):
+        # check if disabled (value set to false)
+        if not config:
+            # Config was set to 'no' instead of yes. Don't do anything then.
+            return
+
+        for entry in task.entries:
+            # If this entry was obtained from the filesystem plugin it should have a filename field. If it does not have
+            # one then there is nothing we can do in this plugin.
+            filename = entry.get('filename')
+            location = entry.get('location')
+
+            # If there is no 'filename' field there is also no nfo file
+            if filename is None or location is None:
+                log.warning("Entry %s didn't come from the filesystem plugin", entry.get('title'))
+                return
+            else:
+                # This will be None if there is no nfo file
+                nfo_filename = self.get_nfo_filename(entry)
+                if nfo_filename is None:
+                    log.warning("Entry %s has no corresponding %s file", entry.get('title'), self.nfo_file_extension)
+                    return
+
+            # Populate the fields from the information in the .nfo file Note that at this point `nfo_filename` has the
+            # name of an existing '.nfo' file
+            self.lookup(entry, nfo_filename)
+
+    def lookup(self, entry, nfo_filename):
+        # If there is already data from a previous parse then we don't need to do anything
+        if entry.get('nfo_id') is not None:
+            log.warning("Entry %s was already parsed by nfo_lookup and it will be skipped. ", entry.get('title'))
+            return
+
+        # nfo_filename Should not be None at this point
+        assert nfo_filename is not None
+
+        # Get all values we can from the nfo file. If the nfo file can't be parsed then a warning is logged and we
+        # return without changing the entry
+        try:
+            nfo_reader = NfoReader(nfo_filename)
+            fields = nfo_reader.get_fields_from_nfo_file()
+        except BadXmlFile:
+            log.warning("Invalid '.nfo' file for entry %s", entry.get('title'))
+            return
+
+        entry.update(fields)
+
+        # If the IMDB id was found in the nfo file, set the imdb_id field of the entry. This will help the imdb_lookup
+        # plugin to get the correct data if it is also used.
+        if 'nfo_id' in fields:
+            entry.update({'imdb_id': fields['nfo_id']})
```

Comment:
Maybe you should use some regex or whatever to verify that it is a valid imdb_id. There is probably a helper function somewhere in the imdb plugins.